transaction-impl/src/main/java/com/example/auction/transaction/impl/TransactionRepository.java

```diff
@@ -0,0 +1,195 @@
+                                "WHERE (creatorId = ? OR winnerId = ?) AND status = ? " +
+                                "ORDER BY itemId DESC " +
+                                "LIMIT ?",
+                        userId,
+                        userId,
+                        status,
+                        limit
+                )
+                .thenApply(List::stream)
+                .thenApply(rows -> rows.skip(offset))
+                .thenApply(rows -> rows.map(TransactionRepository::toTransactionSummary))
+                .thenApply(transactionSummaries -> transactionSummaries.collect(Collectors.toList()))
+                .thenApply(TreePVector::from);
+    }
+
+    private static TransactionSummary toTransactionSummary(Row transaction) {
+        return new TransactionSummary(
+                transaction.getUUID("itemId"),
+                transaction.getUUID("creatorId"),
+                transaction.getUUID("winnerId"),
+                transaction.getString("itemTitle"),
+                transaction.getString("currencyId"),
+                transaction.getInt("itemPrice"),
+                transaction.get("status", TransactionInfoStatus.class)
+        );
+    }
+
+    private static class TransactionEventProcessor extends ReadSideProcessor<TransactionEvent> {
+        private final CassandraSession session;
+        private final CassandraReadSide readSide;
+
+        private PreparedStatement insertTransactionStatement;
+
+        @Inject
+        public TransactionEventProcessor(CassandraSession session, CassandraReadSide readSide) {
+            this.session = session;
+            this.readSide = readSide;
+        }
+
+        @Override
+        public ReadSideHandler<TransactionEvent> buildHandler() {
+            return readSide.<TransactionEvent>builder("transactionEventOffset")
+                    .setGlobalPrepare(this::createTable)
+                    .setPrepare(tag -> prepareStatements())
+                    .setEventHandler(TransactionEvent.TransactionStarted.class,
+                            e -> insertTransaction(e.getItemId(), e.getTransaction()))
+                    .build();
+        }
+
+        @Override
+        public PSequence<AggregateEventTag<TransactionEvent>> aggregateTags() {
+            return TransactionEvent.TAG.allTags();
+        }
+
+        private CompletionStage<Done> createTable() {
+            return session.executeCreateTable(
+                    "CREATE TABLE IF NOT EXISTS transactionSummary (" +
+                            "itemId timeuuid PRIMARY KEY, " +
+                            "creatorId UUID, " +
+                            "winnerId UUID, " +
+                            "itemTitle text, " +
+                            "currencyId text, " +
+                            "itemPrice int, " +
+                            "status text" +
+                            ") " +
+                            "WITH CLUSTERING ORDER BY (itemId DESC)"
```

Comment:
Thanks @TimMoore, now I don't know where to put the filter (by `userType`: `creator` or `winner`). I was thinking into 3 options:
- Create a Cassandra query: I tried `SELECT * FROM userTransactions WHERE userId = ? AND creatorId = ? ...` to get all transactions of a user where the user is the creator, after executing that query, Cassandra asks for `ALLOW FILTERING` because `creatorId` isn't a partition or cluster key.
- Get all transactions (whether the user is creator or winner) and then manually filter in `TransactionRepository.getTransactionsForUser`. It might be weird mix this with `PaginatedSequence` since we don't know how many transactions has a user as a creator or as a winner, then each page may have different size
- Show all the transactions in `web-gateway`, and hide transactions depending on the filter.